prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -252,6 +253,52 @@ object FHIRBundleHelpers {
         receiver: Receiver,
         shortHandLookupTable: MutableMap<String, String>
     ): List<Extension> {
+        val (observationsToKeep, allObservations) =
+            getFilteredObservations(fhirBundle, receiver, shortHandLookupTable)
+
+        val observationExtensionsToKeep = mutableListOf<Extension>()
+        if (observationsToKeep.size < allObservations.size) {
+            observationsToKeep.forEach {
+                observationExtensionsToKeep.add(
+                    Extension(
+                        conditionExtensionurl,
+                        Reference(it.idBase)
+                    )
+                )
+            }
+        }
+        return observationExtensionsToKeep
+    }
+
+    /**
+     * Filter out observations that pass the condition filter for a [receiver]
+     * The [bundle] and [shortHandLookupTable] will be used to evaluate whether
+     * the observation passes the filter
+     *
+     * @return copy of the bundle with filtered observations removed
+     */
+    fun filterObservations(
+        bundle: Bundle,
+        receiver: Receiver,
+        shortHandLookupTable: MutableMap<String, String>
+    ): Bundle {
+        val (observationsToKeep, allObservations) =
+            getFilteredObservations(bundle, receiver, shortHandLookupTable)
+        val filteredBundle = bundle.copy()
+        val listToKeep = observationsToKeep.map { it.idBase }
+        allObservations.forEach {
+            if (it.idBase !in listToKeep) {
+                filteredBundle.deleteResource(it)
+            }
+        }
+        return filteredBundle
+    }
+
+    private fun getFilteredObservations(
+        fhirBundle: Bundle,
+        receiver: Receiver,
```

Comment:
Instead of receiver being a param here, can we take something like `conditionFilter: String`? Then we can do the same thing in `filterObservations` so won't need to build a Receiver in `TranslationTests.kt`. We could also optionally make the change to `getObservationExtensions` so these are all just dealing with condition filters and not with receivers.